soroban-sdk-macros/src/derive_client.rs

```diff
@@ -170,9 +170,18 @@ pub fn derive_client_impl(crate_path: &Path, name: &str, fns: &[syn_ext::Fn]) ->
                 .map(|t| {
                     let ident = match syn_ext::fn_arg_ident(t) {
                         Ok(ident) => ident,
-                        Err(_) => {
-                            errors.push(Error::new(t.span(), "argument not supported"));
-                            format_ident!("")
+                        Err(e) => {
+                            errors.push(e);
+                            if let FnArg::Typed(pat_type) = t {
+                                if let Pat::Wild(_) = *pat_type.pat.clone() {
+                                    // This will result in a clear compiler error if an underscore is used as an argument.
+                                    format_ident!("_")
+                                } else {
+                                    format_ident!("")
+                                }
+                            } else {
+                                format_ident!("")
+                            }
```

Comment:
There are other types of patterns that if matched against would get the `""` and cause the same panic. I think you can make the identifier `"_"` in all error cases.
```suggestion
                            format_ident!("_")
```

Reply:
I initially didn't do this because I thought it might be a confusing if the parameter name was changed to `_`, but this is still better than the existing error. I applied the suggestion.